packages/common/src/crypto/murmurhash.ts

```diff
@@ -0,0 +1,24 @@
+import { MurmurHash3 } from 'murmurhash-wasm';
+
+export class MurmurHash {
+  public static hash(key: string, seed: number): string {
+    const hash = MurmurHash3.hash32(key, seed);
+    const value = hash.readUInt32BE();
+    const hex = value.toString(16);
+
+    let ret = ''; // store the return value
+
+    for (let i = 0; i < hex.length; i++) {
+      // loop through the hex value and convert numbers to alphabet
+      let sub = hex.slice(i, i + 1);
+      if (sub.charCodeAt(0) > 47 && sub.charCodeAt(0) < 58) {
```

Comment:
use `charAt` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/charAt and Number. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger

Reply:
Here's an alternative approach using map. This way we have a separate testable function `charCodeToAlphabet`, as well as a more declarative approach.

```ts
const charCodeToAlphabet = (char: string) => {
  const code = char.charCodeAt(0);
  if (code > 47 && code < 58) return String.fromCharCode(code + 17);
  else return char;
}

...

return hex
  .split('')
  .map(charCodeToAlphabet)
  .join('');
```

Unsure if we gain much benefit from `isInteger`, since we'd have to try parsing it to one first and we need the character code anyway. I guess encoding might be odd, but isn't JS always UTF-8? Or am I wrong on that?